src/main/java/com/ing/data/cassandra/jdbc/CassandraConnection.java

```diff
@@ -166,6 +183,29 @@ public CassandraConnection(final SessionHolder sessionHolder) throws SQLExceptio
         });
     }
 
+    public CassandraConnection(Session cSession, String currentKeyspace, ConsistencyLevel defaultConsistencyLevel, boolean debugMode) {
+        this.sessionHolder = null;
+        this.connectionProperties = new Properties();
+        this.optionSet = optionSet == null ? lookupOptionSet(null) : optionSet;
```

Comment:
Even it's less verbose, please don't use inline conditionals as per our Checkstyle rules:
```
[WARN] /home/runner/work/cassandra-jdbc-wrapper/cassandra-jdbc-wrapper/src/main/java/com/ing/data/cassandra/jdbc/CassandraConnection.java:189:44: Avoid inline conditionals. [AvoidInlineConditionals]
```